File: frontend/src/types/kaggle.ts
export interface KaggleDatasetItem {
  ref: string;
  title: string;
  size: number;
  last_updated: string;
  download_count: number;
  vote_count: number;
  usability_rating: number;
  file_count: number;
}

export interface KaggleFile {
  name: string;
  size: number;
  creation_date: string;
}

export interface KaggleSearchResponse {
  datasets: KaggleDatasetItem[];
  total: number;
}

export interface KaggleImportRequest {
  owner: string;
  slug: string;
  file_name: string;
  dataset_name?: string;
  max_rows?: number;
}

export interface KaggleImportResponse {
  dataset_id: string;
  name: string;
  table_name: string;
  row_count: number;
  columns: string[];
}

File: backend/app/schemas/kaggle.py
from __future__ import annotations

from pydantic import BaseModel


class KaggleDatasetItem(BaseModel):
    ref: str
    title: str
    size: int
    last_updated: str
    download_count: int
    vote_count: int
    usability_rating: float
    file_count: int


class KaggleFile(BaseModel):
    name: str
    size: int
    creation_date: str


class KaggleImportRequest(BaseModel):
    owner: str
    slug: str
    file_name: str
    dataset_name: str | None = None
    max_rows: int | None = None


class KaggleImportResponse(BaseModel):
    dataset_id: str
    name: str
    table_name: str
    row_count: int
    columns: list[str]
